apps/server/src/modules/tldraw/service/tldraw.ws.service.ts

```diff
@@ -87,34 +116,70 @@ export class TldrawWsService {
 	 * @param {WsSharedDocDo} doc
 	 */
 	public updateHandler(update: Uint8Array, origin, doc: WsSharedDocDo): void {
-		const encoder = encoding.createEncoder();
-		encoding.writeVarUint(encoder, WSMessageType.SYNC);
-		writeUpdate(encoder, update);
-		const message = encoding.toUint8Array(encoder);
-		doc.conns.forEach((_, conn) => {
-			this.send(doc, conn, message);
-		});
+		const isOriginWSConn = doc.connections.has(origin as WebSocket);
+		if (isOriginWSConn) {
+			this.pub
+				.publish(doc.name, Buffer.from(update))
+				.catch((err) => this.logger.warning(new RedisPublishErrorLoggable('document', err as Error)));
+		}
+
+		this.propagateUpdate(update, doc);
 	}
 
+	/**
+	 * @param {AwarenessConnectionsUpdate} connectionsUpdate
+	 * @param {WebSocket | null} wsConnection Origin is the connection that made the change
+	 * @param {WsSharedDocDo} doc
+	 */
+	public awarenessUpdateHandler = (
+		connectionsUpdate: AwarenessConnectionsUpdate,
+		wsConnection: WebSocket | null,
+		doc: WsSharedDocDo
+	): void => {
+		const changedClients = this.manageClientsConnections(connectionsUpdate, wsConnection, doc);
+		const buff = this.prepareAwarenessMessage(changedClients, doc);
+		this.sendAwarenessMessage(buff, doc);
+	};
+
 	/**
 	 * Gets a Y.Doc by name, whether in memory or on disk
 	 *
 	 * @param {string} docName - the name of the Y.Doc to find or create
-	 * @param  {boolean} gc - whether to allow gc on the doc (applies only when created)
 	 * @return {WsSharedDocDo}
 	 */
-	getYDoc(docName: string, gc = true): WsSharedDocDo {
+	public getYDoc(docName: string): WsSharedDocDo {
 		return map.setIfUndefined(this.docs, docName, () => {
-			const doc = new WsSharedDocDo(docName, this, gc);
-			if (this.persistence !== null) {
-				this.persistence.bindState(docName, doc).catch(() => {});
-			}
+			const doc = new WsSharedDocDo(docName, this.gcEnabled);
+			doc.awareness.on('update', (connectionsUpdate: AwarenessConnectionsUpdate, wsConnection: WebSocket | null) =>
+				this.awarenessUpdateHandler(connectionsUpdate, wsConnection, doc)
+			);
+			doc.on('update', (update: Uint8Array, origin) => this.updateHandler(update, origin, doc));
+
+			this.sub
```

Comment:
Without await you leave the promise chains is this reall what you want? 
The same for line 167.

Both can also move to private method to split details + logic and orchestration.

Reply:
it is what we wanted